File: 2021/05/09/main.py
class Solution:
    def isPossible(self, target: List[int]) -> bool:
        s, l = sum(target), len(target)
        target.sort()
        if l == 1:
            return target[0] == 1
        while True:
            a = target[l-1]
            s -= a
            if a == 1 or s == 1:
                return True
            if a < s or a % s == 0:
                return False
            a %= s
            s += a
            target[l-1] = a
            target.sort()

File: 2021/05/09/main.go
package main

import "sort"

func isPossible(target []int) bool {
	sort.Ints(target)
	s, l := 0, len(target)
	for i := 0; i < l-1; i++ {
		s += target[i]
	}
	a := target[l-1]
	if l == 1 {
		return a == 1
	}
	if a == 1 || s == 1 {
		return true
	}
	if a < s || a%s == 0 {
		return false
	}
	a %= s
	s += a
	target[l-1] = a
	return isPossible(target)
}
